File: Python3/Redistribute Characters to Make All Strings Equal.py
#1
class Solution:
    def makeEqual(self, words: List[str]) -> bool:
        fre = len(words)
        m = {}
        for word in words:
            for w in word:
                if w not in m:
                    m[w] = 1
                else:
                    m[w] += 1
        
        for v in m.values():
            if v % fre != 0:
                return False

        return True

#2
class Solution:
    def makeEqual(self, words: List[str]) -> bool:
        joint = ''.join(words)
        set1 = set(joint)
        
        for i in set1 :
            if joint.count(i) % len(words) != 0 : return False 
        return True

File: Golang/Redistribute Characters to Make All Strings Equal.go
func makeEqual(words []string) bool {
    charCount := [26]int{}
    n := len(words)
    for _, word := range words {
        for _, c := range word {
            charCount[c - 'a']++
        }
    }
    for _, c := range charCount {
        if c % n != 0 {
            return false
        }
    }
    return true
}